torch/distributions/constraints.py

```diff
@@ -166,21 +168,45 @@ class _LowerTriangular(Constraint):
     Constrain to lower-triangular square matrices.
     """
     def check(self, value):
-        return (torch.tril(value) == value).min(-1)[0].min(-1)[0]
+        masked_value = value*torch.tril(value.new(*value.shape[-2:]).fill_(1.0))
+        return (masked_value == value).min(-1)[0].min(-1)[0]
```

Comment:
NB it would be faster to flatten last two dims (you don't need contiguous because `==` will always give you a contig tensor) and call `min` once